src/main/java/com/google/cloud/resourcemanager/OperationInfo.java

```diff
@@ -0,0 +1,281 @@
+    done = builder.done;
+    metadata = builder.metadata;
+    response = builder.response;
+    error = builder.error;
+  }
+  /**
+   * If the value is false, it means the operation is still in progress. If true, the operation is
+   * completed, and either error or response is available.
+   */
+  public Boolean getDone() {
+    return done;
+  }
+
+  /** Returns the error result of the operation in case of failure or cancellation. */
+  public Status getError() {
+    return error;
+  }
+
+  /**
+   * Returns the service-specific metadata associated with the operation. It typically contains
+   * progress information and common metadata such as create time.
+   */
+  public Map<String, Object> getMetadata() {
+    return metadata;
+  }
+
+  /**
+   * Returns the server-assigned name, which is only unique within the same service that originally
+   * returns it.
+   */
+  public String getName() {
+    return name;
+  }
+
+  /** Returns the normal response of the operation in case of success. */
+  public Map<String, Object> getResponse() {
+    return response;
+  }
+
+  /** Returns a builder for the current operation. */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    OperationInfo that = (OperationInfo) o;
+    return Objects.equals(done, that.done)
+        && Objects.equals(error, that.error)
+        && Objects.equals(metadata, that.metadata)
+        && Objects.equals(name, that.name)
+        && Objects.equals(response, that.response);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(done, error, metadata, name, response);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("name", getName())
+        .add("done", getDone())
+        .add("metadata", getMetadata())
+        .add("response", getResponse())
+        .add("error", getError())
+        .toString();
+  }
+
+  Operation toProtobuf() {
+    Operation operation = new Operation();
+    operation.setName(name);
+    operation.setDone(done);
+    operation.setMetadata(metadata);
+    if (response != null) {
+      operation.setResponse(response);
+    }
+    if (error != null) {
+      operation.setError(error.toProtobuf());
+    }
+    return operation;
+  }
+
+  /** Creates a {@code OperationInfo} object for the provided operation name. */
+  public static OperationInfo of(String name) {
+    return newBuilder(name).build();
+  }
+
+  /**
+   * Returns a {@code OperationInfo} builder where the operation's name is set to the provided name.
+   */
+  public static Builder newBuilder(String name) {
+    return new Builder(name);
+  }
+
+  static OperationInfo fromProtobuf(Operation operation) {
+    Builder builder = newBuilder(operation.getName());
+    if (operation.getName() != null) {
```

Comment:
why all these null checks? If this is nullable, and it looks like it is, you can just copy the nulls over.